src/Nest/Document/Single/Create/CreateRequestJsonConverter.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using System.Linq;
+using System.Reflection;
+using Newtonsoft.Json;
+
+namespace Nest
+{
+	internal class CreateRequestJsonConverter : JsonConverter
```

Comment:
This is essentially identical to `IndexRequestJsonConverter`.  Can we make `IndexRequestJsonConverter` generic instead?